plugins/woocommerce/src/Blocks/Domain/Services/CheckoutLink.php

```diff
@@ -94,21 +94,70 @@ protected function get_products_from_checkout_link() {
 		$raw_products = array_filter( explode( ',', wc_clean( wp_unslash( $_GET['products'] ?? '' ) ) ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
 		$products     = [];
 
-		foreach ( $raw_products as $product_id_qty ) {
-			if ( strpos( $product_id_qty, ':' ) !== false ) {
-				list( $product_id, $qty ) = explode( ':', $product_id_qty );
+		foreach ( $raw_products as $raw_product ) {
+			$cart_item_data = [];
+
+			if ( strpos( $raw_product, ':' ) !== false ) {
+
+				list( $product_id, $qty, $product_data ) = array_pad( explode( ':', $raw_product, 3 ), 3, false );
+
+				// If the quantity is not-numeric, it may be key=value pairs.
+				if ( ! is_numeric( $qty ) ) {
+					$product_data = $qty;
+					$qty          = 1;
+				}
+
+				if ( $product_data ) {
+
+					// Parse key=value pairs separated by semicolons
+					$pairs = explode( ';', $product_data );
```

Comment:
We could probably simplify this by using `parse_str` https://www.php.net/manual/en/function.parse-str.php That uses `&` instead of `;` but if we could substitute. We can then clean up the resulting array.